lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb

```diff
@@ -0,0 +1,139 @@
+require 'fileutils'
+
+Puppet::Type.type(:bootstrap_rpm).provide(:bootstrap_rpm) do
+
+  commands :rpmbuild => 'rpmbuild'
+
+  def create
+    base_dir
+    copy_sources
+    write_specfile
+    build_rpm
+    copy_rpm
+    link_rpm
+  ensure
+    FileUtils.remove_dir(base_dir) if File.directory?(base_dir)
+  end
+
+  def delete
+  end
+
+  def exists?
+    File.exist?(resource[:symlink]) if resource[:symlink]
```

Comment:
If symlink is not given this evaluates to nil and thus false. I think this is what you meant.
```suggestion
    !resource[:symlink] || File.exist?(resource[:symlink])
```